### PR TITLE
OpenShift pipeline changes

### PR DESCRIPTION
**Pull request by Fosol**
Moving CI/CD related configuration to root /openshift to reflect that it is independent of each project.
Adding GitHub owner file to get default reviewers.
Note this updated pipeline follows conventions for all projects within this solution.
Still have a way to go for a pipeline that does everything for every environment.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# These owners will be the default owners for everything in
+# the repo. Unless a later match takes precedence,
+# @global-owner1 and @global-owner2 will be requested for
+# review when someone opens a pull request.
+* @devinleighsmith @Fosol @asanchezr @jdtoombs
```
Comment (Fosol) [inline .github/CODEOWNERS]: By default we should all now be reviewers for every PR.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 **/sh.exe.stackdump
 bash.exe.stackdump
+*.env
```
Comment (Fosol) [inline .gitignore]: Don't ever include any .env file into source to ensure we never save secrets.

File: openshift/README.md [hunk 3]
```diff
@@ -1,6 +1,28 @@
 # OpenShift environment configuration
```
Comment (Fosol) [inline openshift/README.md]: Location for root OpenShift documentation related to this solution.

File: openshift/jenkins/pims-api/Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+// The solr build depends on the solr-base build (more precisely the solr-base image stream)
+// which only needs to be rebuilt when there are changes to it's associated repository (https://github.com/bcgov/openshift-solr)
+// Therefore, it is unnecessary to build it as part of this pipeline.
+
+def PROJECT_NAMESPACE = 'jcxjin'
+def APP_NAME = 'pims'
+def COMP_NAME = 'api'
+def BUILD_ENV = 'tools'
+def CONTEXT_DIRECTORY = 'backend'
+def ENVIRONMENTS = ['dev', 'test']
```
Comment (Fosol) [inline openshift/jenkins/pims-api/Jenkinsfile]: This Jenkinsfile is for deploying to DEV and TEST optionally.

File: openshift/jenkins/pims-api/Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+def TAG_NAMES = ['dev', 'test' ]
+def WAIT_TIMEOUT = 10 // minutes
+
+// Determine whether there were any changes the files within the project's context directory.
+@NonCPS
+boolean triggerBuild(String contextDirectory) {
```
Comment (Fosol) [inline openshift/jenkins/pims-api/Jenkinsfile]: Would love to find a way to move shared library code to a single file somewhere and import it in.

File: openshift/jenkins/pims-api/Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    } catch (err) {
+      echo "Error: ${err}"
+      echo "Not Deployed ..."
+      currentBuild.result = 'FAILURE'
+    }
+  }
+  else {
+    stage('No Changes') {
+      echo "No changes ..."
+      currentBuild.result = 'ABORTED'
```
Comment (Fosol) [inline openshift/jenkins/pims-api/Jenkinsfile]: Aborted releases are not stored in the list!  It's so clean now.

File: openshift/jenkins/utils.groovy [hunk 7]
```diff
@@ -0,0 +1,72 @@
+package pims.utils
```
Comment (Fosol) [inline openshift/jenkins/utils.groovy]: The idea is to use this a shared library.  But I don't know how to import it.

**Comment by devinleighsmith** [comment]
@Fosol Can you rebase to reduce the number of commits in this PR? There are 65 commits in this PR, a large number of which are just "Fixing x". As much as I like the idea of small commits, having this many commits for a story is going to make navigating the git commit history a problem. In addition I don't think there is a benefit to seeing all the commits where you were figuring things out, just the final commits where you did figure the thing out

File: openshift/jenkins/pims-app/Jenkinsfile [hunk 8]
```diff
@@ -0,0 +1,173 @@
+// Implementation Details
```
Comment (devinleighsmith) [inline openshift/jenkins/pims-app/Jenkinsfile]: Does this replace frontend/jenkinsfile? if so feel free to remove that file.

File: openshift/templates/app/pims-app/build.yaml [hunk 9]
```diff
@@ -0,0 +1,181 @@
+kind: Template
```
Comment (devinleighsmith) [inline openshift/templates/app/pims-app/build.yaml]: Can this replace the corresponding file within /frontend/openshift ?

File: openshift/jenkins/pims-app/Jenkinsfile (hunk 8 above)
Comment (Fosol) [reply openshift/jenkins/pims-app/Jenkinsfile]: I didn't want to remove your files until this was working after the merge.

**Pull request closed without merging**